operator/builtin/output/newrelic/newrelic.go

```diff
@@ -0,0 +1,177 @@
+	Timeout    helper.Duration `json:"timeout,omitempty"     yaml:"timeout,omitempty"`
+}
+
+// Build will build a new NewRelicOutput
+func (c NewRelicOutputConfig) Build(context operator.BuildContext) (operator.Operator, error) {
+	outputOperator, err := c.OutputConfig.Build(context)
+	if err != nil {
+		return nil, err
+	}
+
+	headers, err := c.getHeaders()
+	if err != nil {
+		return nil, err
+	}
+
+	buffer, err := c.BufferConfig.Build(context, c.ID())
+	if err != nil {
+		return nil, err
+	}
+
+	url, err := url.Parse(c.BaseURI)
+	if err != nil {
+		return nil, errors.Wrap(err, "parse 'base_uri'")
+	}
+
+	nro := &NewRelicOutput{
+		OutputOperator: outputOperator,
+		buffer:         buffer,
+		client:         &http.Client{},
+		headers:        headers,
+		url:            url,
+		timeout:        c.Timeout.Raw(),
+	}
+
+	nro.flusher = c.FlusherConfig.Build(buffer, nro.ProcessMulti, nro.SugaredLogger)
+
+	return nro, nil
+}
+
+func (c NewRelicOutputConfig) getHeaders() (http.Header, error) {
+	headers := http.Header{
+		"X-Event-Source":   []string{"logs"},
+		"Content-Encoding": []string{"gzip"},
+	}
+
+	if c.APIKey == "" && c.LicenseKey == "" {
+		return nil, fmt.Errorf("one of 'api_key' or 'license_key' is required")
+	} else if c.APIKey != "" {
+		headers["X-Insert-Key"] = []string{c.APIKey}
+	} else {
+		headers["X-License-Key"] = []string{c.LicenseKey}
+	}
+
+	return headers, nil
+}
+
+// NewRelicOutput is an operator that sends entries to the New Relic Logs platform
+type NewRelicOutput struct {
+	helper.OutputOperator
+	buffer  buffer.Buffer
+	flusher *flusher.Flusher
+
+	client  *http.Client
+	url     *url.URL
+	headers http.Header
+	timeout time.Duration
+}
+
+// Start tests the connection to New Relic and begins flushing entries
+func (nro *NewRelicOutput) Start() error {
+	// Test connection
+	ctx, cancel := context.WithTimeout(context.Background(), nro.timeout)
+	defer cancel()
+	if err := nro.ProcessMulti(ctx, nil); err != nil {
+		return errors.Wrap(err, "test connection")
+	}
+
+	nro.flusher.Start()
+	return nil
+}
+
+// Stop tells the NewRelicOutput to stop gracefully
+func (nro *NewRelicOutput) Stop() error {
+	nro.flusher.Stop()
+	return nro.buffer.Close()
+}
+
+// Process adds an entry to the output's buffer
+func (nro *NewRelicOutput) Process(ctx context.Context, entry *entry.Entry) error {
+	return nro.buffer.Add(ctx, entry)
+}
+
+// ProcessMulti will send a chunk of entries to New Relic
+func (nro *NewRelicOutput) ProcessMulti(ctx context.Context, entries []*entry.Entry) error {
+	lp := LogPayloadFromEntries(entries)
+
+	ctx, cancel := context.WithTimeout(ctx, nro.timeout)
+	defer cancel()
+	req, err := nro.newRequest(ctx, lp)
+	if err != nil {
+		return errors.Wrap(err, "create request")
+	}
+
+	res, err := nro.client.Do(req)
+	if err != nil {
+		return errors.Wrap(err, "execute request")
+	}
+	if !(res.StatusCode >= 200 && res.StatusCode < 300) {
```

Comment:
nit: Should anything other than 2xx really cause an error?

There are some cases where this might be unnecessary. Maybe we should just log a warning and continue in some cases.

Reply:
I also wondered about this, but it appears the API is pretty basic, and anything other than a 2xx indicates a failure. I pulled this behavior directly from the [FluentD output plugin](https://github.com/newrelic/newrelic-fluentd-output/blob/74bebf7be25ed1cadb10ea61fa3de234b4e1f489/lib/fluent/plugin/out_newrelic.rb#L111) which is maintained by New Relic

Reply:
Actually, looking closer, the NR plugin just logs an error, but doesn't return one. So maybe we shoudl do the same